src/conv-core/cpuaffinity.C

```diff
@@ -442,62 +453,226 @@ static void cpuAffinityHandler(void *m)
 */
   CmiAssert(CmiMyPe()==0 && rankmsg != NULL);
   skt_ip_t & ip = msg->ip;
-  pe = msg->pe;
+  int node = msg->node;
+  int pe = msg->pe;
   auto iter = hostTable.find(ip);
   if (iter != hostTable.end()) {
     rec = iter->second;
     CmiFree(msg);
   }
   else {
     rec = msg;
-    rec->seq = nodecount;
-    nodecount++;                          /* a new node record */
+    rec->seq = hostcount++;
     hostTable.emplace(ip, msg);
   }
-  myrank = rec->rank%rec->ncores;
-  while (in_exclude(myrank)) {             /* skip excluded core */
-    myrank = (myrank+1)%rec->ncores;
-    rec->rank ++;
+
+  int & node_host = rankmsg->node_hosts[node];
+  if (node_host == -1)
+  {
+    rankmsg->node_ranks_on_host[node] = rec->node_rank++;
+    node_host = rec->seq;
```

Comment:
`node_host` appears to be unused. I think it'd be clearer if the assignment done to `node_hosts[node]` was done explicitly rather than via a reference.

Reply:
`node_host` is checked against -1 and then assigned. I made it a reference because of these two uses.

Reply:
Yeah, I see that, I'm fine with it as it is, I just think I stylistically prefer not doing it like that because it's less clear to me that the assignment is really happening in `node_hosts`.